Sources/GRPCReflectionService/Documentation.docc/ReflectionServiceTutorial.md

```diff
@@ -0,0 +1,235 @@
+- [Sources/GRPCReflectionService/v1/reflection-v1.proto][v1]
+
+Depending on the version of gRPCurl you are using you might need to use the v1alpha proto instead.
+
+ ### Instantiating the Reflection Service 
+
+ In the Server implementation, we have to instantiate each provider.
+ For instantiating the `ReflectionService` provider we have to pass in an array
+ of Strings representing the paths to the binary files we have just generated,
+withing the project and the version of the reflection, in our case v1.
+
+
+```swift
+let paths = [
+  "Sources/Examples/ReflectionService/Generated/helloworld.grpc.reflection.txt", 
+  "Sources/Examples/ReflectionService/Generated/echo.grpc.reflection.txt", 
+  "Sources/Examples/ReflectionService/Generated/reflection.grpc.reflection.txt"
+  ]
+
+let reflectionServiceProvider = try ReflectionService(
+  serializedFileDescriptorProtoFilePaths: paths, version: .v1
+)
+```
+
+### Running the Server
+
+In our example the server is not configured with TLS. The port is `1234`.
+Starting the server:
+
+```swift
+let server = try await Server.insecure(group: group)
+  .withServiceProviders([reflectionServiceProvider, GreeterProvider(), EchoProvider()])
+  .bind(host: "localhost", port: self.port)
+  .get()
+
+```
+
+To run the server, from the root of the package run:
+
+```sh
+$ swift run ReflectionServiceServer
+```
+
+ ## Testing the Reflection Service using GRPCurl
+
+Please follow the instructions from the [GRPCurl README][grpcurl-setup] in order to set gRPCurl up.
+
+From a different terminal than the one used for running the Server, we will call gRPCurl commands,
+following the format:
+
+`grpcurl [flags] [address] [list|describe] [symbol]`.
+
+In our case we are using the `-plaintext` flag, because our server isn't configured with TLS, and 
+the address is set to `localhost:1234`.
+
+Here are some gRPCurl commands and the responses:
+
+- **List services**
+```sh
+$ grpcurl -plaintext localhost:1234 list
+```
+
+**output:**
+```sh
+Echo
+Greeter
+ServerReflection
+```
+
+- **List methods of a service**
+```sh
+$ grpcurl -plaintext localhost:1234 list echo.Echo
+```
+
+**output:**
+```sh
+echo.Echo.Collect
+echo.Echo.Expand
+echo.Echo.Get
+echo.Echo.Update
+```
+
+- **Describe a service**
+```sh
+$ grpcurl -plaintext localhost:1234 describe echo.Echo
+```
+
+**output:**
+```
+echo.Echo is a service:
+service Echo {
+  // Collects a stream of messages and returns them concatenated when the caller closes.
+  rpc Collect ( stream .echo.EchoRequest ) returns ( .echo.EchoResponse );
+  // Splits a request into words and returns each word in a stream of messages.
+  rpc Expand ( .echo.EchoRequest ) returns ( stream .echo.EchoResponse );
+  // Immediately returns an echo of a request.
+  rpc Get ( .echo.EchoRequest ) returns ( .echo.EchoResponse );
+  // Streams back messages as they are received in an input stream.
+  rpc Update ( stream .echo.EchoRequest ) returns ( stream .echo.EchoResponse );
+}
+```
```

Comment:
```suggestion
You can also get descriptions of objects like services, methods, and messages. The following
command fetches a description of the Echo service:
 
```sh
$ grpcurl -plaintext localhost:1234 describe echo.Echo
echo.Echo is a service:
service Echo {
  // Collects a stream of messages and returns them concatenated when the caller closes.
  rpc Collect ( stream .echo.EchoRequest ) returns ( .echo.EchoResponse );
  // Splits a request into words and returns each word in a stream of messages.
  rpc Expand ( .echo.EchoRequest ) returns ( stream .echo.EchoResponse );
  // Immediately returns an echo of a request.
  rpc Get ( .echo.EchoRequest ) returns ( .echo.EchoResponse );
  // Streams back messages as they are received in an input stream.
  rpc Update ( stream .echo.EchoRequest ) returns ( stream .echo.EchoResponse );
}
```